src/software/ai/motion_constraint/motion_constraint_set_builder.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <set>
+
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/motion_constraint/motion_constraint.h"
+#include "software/ai/motion_constraint/motion_constraint_visitor.h"
+#include "software/world/game_state.h"
+
+class MotionConstraintSetBuilder
+{
+   public:
+    explicit MotionConstraintSetBuilder() = default;
+
+    /**
+     * Builds Motion Constraint based on gamestate and tactic
+     *
+     * @param gamestate Current GameState to process
+     * @param tactic Current Tactic to process
+     *
+     * @return set of MotionConstraints
+     */
+    std::set<MotionConstraint> buildMotionConstraintSet(const GameState &game_state,
+                                                        Tactic &tactic);
+
+   private:
+    std::set<MotionConstraint> current_allowed_constraints;
+    MotionConstraintVisitor motion_constraint_visitor;
+
+    /**
+     * Adds move constraints determined from gamestate to current_motion_constraints
```

Comment:
This function seems to return the motion contraint set now, looks like the comment needs to be updated